spirv_msl.cpp

```diff
@@ -11033,10 +11042,21 @@ void CompilerMSL::analyze_argument_buffers()
 			}
 			else if (!constexpr_sampler)
 			{
-				// constexpr samplers are not declared as resources.
-				add_resource_name(var_id);
-				resources_in_set[desc_set].push_back(
-				    { &var, to_name(var_id), type.basetype, get_metal_resource_index(var, type.basetype), 0 });
+				if (msl_options.is_ios() && type.basetype == SPIRType::Image && type.image.sampled == 2)
+				{
+					// Very specifically, image load-store in argument buffers are disallowed on MSL on iOS.
+					// But we won't know when the argument buffer is encoded whether this image will have
+					// a NonWritable decoration. So just use discrete arguments for all storage images
+					// on iOS.
+					discrete_storage_images.insert(var_id);
```

Comment:
Going through a std::unordered_set causes MSVC to emit different code. get_metal_resource_index() is called in flipped order, so the resources get index 1, 0 instead of 0, 1. Is it possible to check for this twice with positive test and negative test instead of going through a map?